sdk/eventhub/event-hubs/src/eventDataBatch.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventData, toAmqpMessage } from "./eventData";
+import { ConnectionContext } from "./connectionContext";
+import { AmqpMessage } from "@azure/core-amqp";
+import { message } from "rhea-promise";
+
+/**
+ * A class representing a batch of events which can be passed to the `send` method of a `EventConsumer` instance.
+ * This batch is ensured to be under the maximum size limited supported by Azure Event Hubs service.
+ *
+ * Use the `tryAdd` function on the EventDataBatch to add events in a batch.
+ * @class
+ */
+export class EventDataBatch {
+  /**
+   * @property Describes the amqp connection context for the Client.
+   */
+  private _context: ConnectionContext;
+  /**
+   * @property Total length of events.
+   */
+  private _eventsLength: number;
```

Comment:
The only place `_eventsLength` seems to be used is when an encoded message is added to `this._encodedMessages`. That code can be updated to use `this._encodedMessages.push()` and we can remove the need for `_eventsLength`